dfusion_rust_core/src/models/standing_order.rs

```diff
@@ -1,24 +1,46 @@
-use serde_derive::{Deserialize};
-use sha2::{Digest, Sha256};
-use web3::types::{H256, U256};
-use crate::models::{ConcatenatingHashable, RollingHashable};
 use crate::models;
+use crate::models::{ConcatenatingHashable, RollingHashable};
+
 use array_macro::array;
+use serde_derive::Deserialize;
+use sha2::{Digest, Sha256};
+use std::sync::Arc;
+use web3::types::{Log};
+use web3::types::{H256, U256};
+use graph::data::store::{Entity};
 
+use super::util::*;
 
 #[derive(Debug, Clone, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
 #[serde(rename_all = "camelCase")]
 pub struct StandingOrder {
     pub account_id: u16,
     pub batch_index: U256,
+    pub valid_from_auction_id: U256,
     orders: Vec<super::Order>,
 }
+pub struct EncodedOrder {
```

Comment:
Not sure it's worthwhile creating a public struct for this. It might be a bit confusing what it is used for, when really it's just a way to convert aggregated information we receive in a Log to an order.

I'd just create a method on `Order` directly that is something like `from_standing_order_event(account_id: u16, ..., bytes: Vec<u8>)`

Reply:
https://github.com/gnosis/dex-services/pull/211/commits/9b3d3412703f4263408bb98e27d484885942cfbd